File: _daily_topic/slidingWindow/1838.cpp
#include "header.h"
//最高频元素的频数
class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end()); 
        int left = 0, right = 1;
        int n = nums.size();
        long total = 0;
        int res = 1;
        while (right < n) {
            total += (long)(nums[right] - nums[right - 1]) * (right - left);
            while (total > k) {
                total -= (nums[right] - nums[left]);
                ++left;
            }
            res = max(res, right - left + 1);
            ++right;
        }
        return res;
    }
};

File: _daily_topic/slidingWindow/1838.py
from typing import List
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        n = len(nums)
        left, right, total, res = 0, 1, 0, 1
        while right < n :
            total = total + (nums[right] - nums[right - 1]) * (right - left)
            while total > k :
                total = total - (nums[right] - nums[left])
                left = left + 1
            res = max(res, right - left + 1)
            right = right + 1
        return res